src/main/java/org/broadinstitute/consent/http/service/UserService.java

```diff
@@ -342,6 +368,29 @@ public void insertUserRoles(List<UserRole> roles, Integer userId) {
         userRoleDAO.insertUserRoles(roles, userId);
     }
 
+    private Integer getSigningOfficialForUser(Integer userId) {
+        List<UserProperty> props =
+                userPropertyDAO.findUserPropertiesByUserIdAndPropertyKeys(
+                        userId,
+                        List.of(UserFields.SELECTED_SIGNING_OFFICIAL_ID.getValue())
+                );
+
+        if (props.size() == 0) {
+            return null;
+        }
+
+        UserProperty soIdProp = props.get(0);
+
+        int soId;
+        try {
+            soId = Integer.parseInt(soIdProp.getPropertyValue());
+        } catch (NumberFormatException e) {
+            return null;
+        }
```

Comment:
I think we have another error case here which is if we don't have an actual user matching the id.